packages/theming/src/utils/getColor.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { default as defaultTheme } from '../theme';
+import { darken, lighten } from 'polished';
+
+const toHue = (hue, theme) => {
+  const colors = theme && theme.colors ? theme.colors : defaultTheme.colors;
+
+  switch (hue) {
+    case '__primary':
+      return colors.primaryHue;
+    case '__danger':
+      return colors.dangerHue;
+    case '__warning':
+      return colors.warningHue;
+    case '__success':
+      return colors.successHue;
+    case '__neutral':
+      return colors.neutralHue;
+    case '__chrome':
+      return colors.chromeHue;
+    default:
+      return hue || colors.primaryHue;
+  }
+};
+
+/**
+ * Get the palette color for the given hue, shade, and theme.
+ *
+ * @param {string|Object} [props.hue] A palette hue or one of the following reserved keys:
+ *  - `'__primary'` = `theme.colors.primaryHue`
+ *  - `'__danger'` = `theme.colors.dangerHue`
+ *  - `'__warning'` = `theme.colors.warningHue`
+ *  - `'__success'` = `theme.colors.successHue`
+ *  - `'__neutral'` = `theme.colors.neutralHue`
+ *  - `'__chrome'` = `theme.colors.chromeHue`
+ * @param {number} [props.shade=600] A hue shade.
+ * @param {Object} [props.theme] Context `theme` object.
+ *
+ * @component
+ */
+export default function getColor({ hue, shade = 600, theme } = {}) {
```

Comment:
Would it be possible for this function to be used for retrieving the `foreground` and `background` colors? Might need an override without a `shade`, but I think it might be helpful for all color usages to go through this utility.